src/Utilities/MicronTrackerInterface/Cameras.cpp

```diff
@@ -11,89 +11,83 @@
 *
 ***************************************************************/
 
-#include "MTC.h"
-
 #include "Cameras.h"
 #include "MCamera.h"
-//#include "MTVideo.h"
 #include "MicronTrackerLoggerMacros.h"
 
-/****************************/
-/** Constructor */
+//------------------------------------------
 Cameras::Cameras()
 {
-  this->ownedByMe = TRUE;
-  this->mCurrCam = NULL;
-  this->mFailedCam = NULL;
+  this->OwnedByMe = true;
+  this->CurrentCamera = NULL;
+  this->LastFailedCamera = NULL;
   Cameras_HistogramEqualizeImagesSet(true);
   // error handling here
 }
 
-/****************************/
-/** Destructor */
+//------------------------------------------
 Cameras::~Cameras()
 {
   // Clear all previously connected camera
-  for (std::vector<MCamera *>::iterator camsIterator = m_vCameras.begin(); camsIterator != m_vCameras.end(); ++camsIterator)
+  for (std::vector<MCamera*>::iterator camsIterator = CameraList.begin(); camsIterator != CameraList.end(); ++camsIterator)
   {
     free(*camsIterator);
   }
-  m_vCameras.clear();
+  CameraList.clear();
 
-  if (mCurrCam != NULL)
+  if (CurrentCamera != NULL)
   {
-    free(mCurrCam);
-    mCurrCam=NULL;
+    free(CurrentCamera);
+    CurrentCamera = NULL;
   }
 
-  if (mFailedCam != NULL) 
+  if (LastFailedCamera != NULL)
   {
-    free(mFailedCam);
-    mFailedCam=NULL;
+    free(LastFailedCamera);
+    LastFailedCamera = NULL;
   }
 }
 
-/****************************/
-/** */ 
+//------------------------------------------
 bool Cameras::getHistogramEqualizeImages()
 {
-	bool R;
-	Cameras_HistogramEqualizeImagesGet(&R);
-	return R;
+  bool R;
+  Cameras_HistogramEqualizeImagesGet(&R);
+  return R;
 }
 
-/****************************/
-int Cameras::setHistogramEqualizeImages(bool on_off)
+//------------------------------------------
+bool Cameras::setHistogramEqualizeImages(bool on_off)
 {
-	int result = Cameras_HistogramEqualizeImagesSet(on_off);

```

Comment:
This was not a Boolean return value but an error code. By changing it to Boolean we don't know what was the exact failure.

Reply:
Well, that's a pattern over the whole API. The int return type was actually the #define's in MTC.h

If you want to keep that API behaviour, I can wrap those defines in an enum and return that instead.

Reply:
My point is that there is no need to leak out MTC.h through the wrapper's interface if there are methods the wrapper that give info about the error code.

If MTC.h only contains these error codes then it may be fine to make it available outside, but preferably the wrapper should hide all internals.

Reply:
hmm, no, my solution only works with c++11